sparkle/output/configuration_output.py

```diff
@@ -0,0 +1,243 @@
+        self.directory = path
+
+        if output.is_dir():
+            self.output = output / "configuration.json"
+        else:
+            self.output = output
+
+        # TODO: Fix this spaghetti code to find the scenario file
+        solver_dir_name = path.name
+        scenario_dir = path / "outdir_train_configuration" / \
+            f"{solver_dir_name}_scenario" / "state-run0"
+        if scenario_dir.is_dir():
+            scenario_file = scenario_dir / "scenario.txt"
+            if not scenario_file.is_file():
+                print("Can't find scenario file")
+                return
+        else:
+            print("Can't find scenario file")
+            return
+
+        scenario_dir = scenario_dir.parent
+        # Sets scenario on configurator object
+        self.configurator.scenario = \
+            ConfigurationScenario.from_file(scenario_file, self.solver,
+                                            self.instance_set_train,
+                                            scenario_dir)
+        self.configurator.scenario._set_paths(self.configurator.output_path)
+
+        # Retrieve all configurations
+        config_path = path / "validation" / "configurations.csv"
+        self.configurations = self.get_configurations(config_path)
+
+        # Retrieves validation results and best configuration
+        self.training, self.best_config =\
+            self.get_validation_data(self.instance_set_train)
+
+        # Retrieve test validation results if they exist
+        if self.instance_set_test is not None:
+            self.test, _ = self.get_validation_data(self.instance_set_test)
+
+    def get_configurations(self: ConfigurationOutput, config_path: Path) -> list[dict]:
+        """Read all configurations and transform them to dictionaries."""
+        configs = []
+
+        # TODO: Should we check if the configurations are unique?
+        # Check if the path exists and is a file
+        if config_path.exists() and config_path.is_file():
+            with config_path.open("r") as file:
+                for line in file:
+                    config = Solver.config_str_to_dict(line.strip())
+                    configs.append(config)
+
+        else:
+            print("Can't find configurations")
+
+        return configs
+
+    def get_validation_data(self: ConfigurationOutput, instance_set: InstanceSet) -> \
+            tuple[ConfigurationResults, str]:
+        """Returns best config and ConfigurationResults for instance set."""
+        objective = self.configurator.scenario.sparkle_objective
+
+        # Retrieve found configuration
+        x, best_config = self.configurator.get_optimal_configuration(
+            self.solver, instance_set, objective.PerformanceMeasure)
+        # Will be removed once I know what to do with x
+        print(x)
+
+        # Retrieve validation results
+        validator = Validator(self.directory)
+        val_default = validator.get_validation_results(
```

Comment:
In this case it looks like you only retrieve the validation results of the default and the best configurations. I think this object should be more general and get **all** found configurations and **all** of their validation results. Then the report generator can take in all this information and determine which data to show in the report